metadata/src/main/java/org/apache/kafka/metadata/PartitionRegistration.java

```diff
@@ -337,6 +338,15 @@ public int preferredReplica() {
         return replicas.length == 0 ? LeaderConstants.NO_LEADER : replicas[0];
     }
 
+    public Uuid directory(int replica) {
+        Map<Integer, Uuid> assignment = DirectoryId.createAssignmentMap(replicas, directories);
```

Comment:
I don't think it makes sense to create a whole map just for this function, then throw it away

If we're going to iterate over all the replicas, just find the index of the one that was passed in, and then return `directories[index]`